plugins/outputs/whatap/DataOutputX.go

```diff
@@ -0,0 +1,204 @@
+package whatap
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+)
+
+const (
+	INT3_MIN_VALUE  = -8388608 /*0xff800000*/
+	INT3_MAX_VALUE  = 0x007fffff
+	LONG5_MIN_VALUE = -549755813888 /*0xffffff8000000000*/
+	LONG5_MAX_VALUE = 0x0000007fffffffff
+)
+
+type DataOutputX struct {
+	buffer  bytes.Buffer
+	written int
+}
+
+func NewDataOutputX() *DataOutputX {
+	in := new(DataOutputX)
+	return in
+}
+
+func (out *DataOutputX) ToByteArray() []byte {
+	return out.buffer.Bytes()
+}
+
+func (out *DataOutputX) WriteIntBytes(b []byte) *DataOutputX {
+	if b == nil || len(b) == 0 {
+		out.WriteInt(0)
+	} else {
+		out.WriteInt(int32(len(b)))
+		out.WriteBytes(b)
+	}
+	return out
+}
+
+func (out *DataOutputX) WriteBlob(value []byte) *DataOutputX {
+	if value == nil || len(value) == 0 {
+		out.WriteByte1(0)
+	} else {
+		sz := len(value)
+		if sz <= 253 {
+			out.WriteByte1(byte(sz))
+			out.WriteBytes(value)
+		} else if sz <= 65535 {
+			buff := []byte{255, 0, 0}
+			out.WriteBytes(SetBytesShort(buff, 1, int16(sz)))
+			out.WriteBytes(value)
+		} else {
+			buff := []byte{254, 0, 0, 0, 0}
+			out.WriteBytes(SetBytesInt(buff, 1, int32(sz)))
+			out.WriteBytes(value)
+		}
+	}
+	return out
+}
+func (out *DataOutputX) WriteDecimal(v int64) *DataOutputX {
+
```

Comment:
Please remove those newlines at the begin of the function body. Only use newlines to structure the code.

Reply:
Moved to a library outpside of telegraf